src/main/java/io/github/filelize/Filelizer.java

```diff
@@ -32,78 +32,63 @@ public Filelizer(String basePath) {
     }
 
     public Filelizer(String basePath, ObjectMapper objectMapper, FilelizeType defaultFilelizeType) {
-        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.findAndRegisterModules();
         objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
         this.filelizerObject = new FilelizerObject(basePath, objectMapper);
         this.filelizerSingle = new FilelizerSingle(basePath, objectMapper);
         this.filelizerMultiple = new FilelizerMultiple(basePath, objectMapper);
         this.defaultFilelizeType = defaultFilelizeType;
     }
 
+    private IFilelizer resolveFilelizerType(Object valueType) {
+        return resolveFilelizerType(getFilelizeType(valueType, defaultFilelizeType));
+    }
+
+    private IFilelizer resolveFilelizerType(FilelizeType filelizeType) {
```

Comment:
This is a nice cleanup. I've just updated to java 17 so you should be able to change the syntax.